phangsPipeline/casaVisRoutines.py

```diff
@@ -1600,361 +1688,524 @@ def reweight_data(
+        infile_list=split_file_list, outfile=outfile, freqtol=freqtol,
+        overwrite=overwrite, copypointing=(not clear_pointing))
 
     # Clean up, deleting intermediate files
 
     for this_file in split_file_list:
         shutil.rmtree(this_file)
-    
+
     # Remove touch file to mark that we are have done the processing of this data
     if os.path.isdir(outfile+'.touch'):
         os.rmdir(outfile+'.touch')
-    
+
     return()
 
+
 ########################################
 # Extract a continuum measurement set. #
 ########################################
-                      
+
+
 def extract_continuum(
-    infile = None, 
-    outfile = None, 
-    ranges_to_exclude = [],    
-    lines_to_flag = [],
-    vsys_kms=None, vwidth_kms=None, 
-    vlow_kms=None, vhigh_kms=None,    
-    do_statwt = False, 
-    do_collapse = True, 
-    overwrite = False, 
-    ):
+        infile=None, outfile=None, spw=None,
+        range_to_extract=None, method='regrid_then_rebin',
+        target_chan_ghz=None, binfactor=None, nchan=None, lines_to_flag=None,
+        vlow_kms=None, vhigh_kms=None, vsys_kms=None, vwidth_kms=None,
+        require_full_cont_coverage=False, overwrite=False):
     """
-    Extract continuum uv data from a measurement set. Optionally
-    reweights and collapses the data to a single channel per spw.
-        
-    Args:
-    
-    infile (str): The input measurement set data with suffix ".ms".
-    outfile (str): The output measurement set data with suffix ".ms".
-    do_statwt (bool): 
-    do_collapse (bool): Always True to produce the single-channel continuum data.
-    
-    Inputs:
-    
-    Outputs:
-    
+    Continuum extraction routine. Takes infile, outfile, ranges of interest,
+    lines to flag, and algorithm, along with algorithm tuning parameters.
     """
-    
-    # Check input
 
+    # Check the method
+    valid_methods = [
+        'regrid_then_rebin', 'rebin_then_regrid', 'just_regrid', 'just_rebin']
+    if method.lower().strip() not in valid_methods:
+        logger.error("Not a valid continuum extraction method - "+str(method))
+        raise Exception("Please specify a valid continuum extraction method.")
+
+    # Check input
     if infile is None:
         logging.error("Please specify an input file.")
         raise Exception("Please specify an input file.")
 
+    if not os.path.isdir(infile):
+        logger.error(
+            'The input measurement set "'+infile+'"does not exist.')
+        raise Exception(
+            'The input measurement set "'+infile+'"does not exist.')
+
     if outfile is None:
         logging.error("Please specify an output file.")
         raise Exception("Please specify an output file.")
 
-    if not os.path.isdir(infile):
-        logger.error('The input measurement set "'+infile+'"does not exist.')
-        raise Exception('The input measurement set "'+infile+'"does not exist.')
+    # if range_to_extract is None:
```

Comment:
Why is this commented out? I don't see a non-None default being set so I think this should be included.

Reply:
But the diff is odd here so maybe it was like this before?

Reply:
It looks like @thomaswilliamsastro added an if condition to deal with the `range_to_extract = None` case [below](https://github.com/akleroy/phangs_imaging_scripts/blob/dea4d38763b8644e4fdd2469957e4bffe5ab90a9/phangsPipeline/casaVisRoutines.py#L1974)

Reply:
So we can probably delete these commented lines already.

Reply:
Yep, that was me leaving stuff in in case I broke it. Gone now!